### PR TITLE
[component] Rename "In development" stability level to "Development"

### PR DESCRIPTION
**Pull request by dmitryax**
The decision to rename this stability level was made in https://github.com/open-telemetry/opentelemetry-collector/pull/6531#discussion_r1020619947

### CODE REVIEW

**Review by codeboten — APPROVED** [review]

**Comment by dmitryax** [comment]
To fix failing contrib-tests https://github.com/open-telemetry/opentelemetry-collector/pull/6563

File: README.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ we intend to provide high-quality components as part of this repository, we ackn
 for prime time. As such, each component should list its current stability level for each telemetry signal, according to
 the following definitions:
 
-### In development
+### Development
```
Comment (TylerHelmuth) [inline README.md:74]: This rename will break a lot of links in contrib right?

Comment (dmitryax) [reply README.md:74]: Yes, I'll fix them once this is merged

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Look at the errors:

```
Error: /home/runner/work/opentelemetry-collector/opentelemetry-collector/exporter/loggingexporter/factory.go:46:64: undefined: component.StabilityLevelDevelopment
Error: /home/runner/work/opentelemetry-collector/opentelemetry-collector/exporter/loggingexporter/factory.go:47:66: undefined: component.StabilityLevelDevelopment
Error: /home/runner/work/opentelemetry-collector/opentelemetry-collector/exporter/loggingexporter/factory.go:48:60: undefined: component.StabilityLevelDevelopment
```

Something is wrong with cross links

**Comment by bogdandrutu** [comment]
Needs a rebase

**Comment by dmitryax** [comment]
Rebased

**Pull request merged**